src/ckb_chain/funding/funding_tx.rs

```diff
@@ -200,10 +301,28 @@ impl FundingTxBuilder {
             CapacityBalancer::new_simple(sender, placeholder_witness, self.request.local_fee_rate);
 
         let ckb_client = CkbRpcClient::new(&self.context.rpc_url);
-        let cell_dep_resolver = {
+        let mut cell_dep_resolver = {
             let genesis_block = ckb_client.get_block_by_number(0.into()).unwrap().unwrap();
             DefaultCellDepResolver::from_genesis(&BlockView::from(genesis_block)).unwrap()
         };
+
+        if let Some(ref udt_info) = self.request.udt_info {
+            // FIXME(yukang): how to add cell deps for udt?
+            let udt_type_script = udt_info.type_script.clone();
```

Comment:
For devnet and mocknet (a mocked chain environment), we can find cell deps for udt just like finding cell deps for commitment lock. See https://github.com/contrun/ckb-pcn-node/blob/1b22666d295f8a8a4b07c0035a681bf1c6907ce7/src/ckb_chain/contracts.rs#L28 for mocknet and https://github.com/contrun/ckb-pcn-node/blob/1b22666d295f8a8a4b07c0035a681bf1c6907ce7/src/ckb_chain/contracts.rs#L231 for devnet. For testnet and mainnet later, we may hard code these transaction hashes.

Reply:
add this commit: https://github.com/contrun/ckb-pcn-node/pull/37/commits/191bb361284ad95fa9cd94871a2de15bf634305c